### ISSUE TITLE
Category - No hover effect for an already selected category

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#39716]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___
 
**Version Number:** 1.4.60-1
**Reproducible in staging?:** y
**Reproducible in production?:** y
**Issue reported by:** Applause - Internal Team

## Action Performed:
Precondition:
- Create a collect workspace on OD and enable chat for newDot
- Go to tags settings on OD for the collect workspace and add a few tags
1. Navigate to staging.new.expensify.com
2. Open workspace chat page
3. Start a manual IOU flow and select a category
4. Click on tag and select one
5. Click on category
6. Press the keyboard down arrow a few times to shift the focus
7. Hover over the selected category
8. Go back and click on tags
9. Press the keyboard down arrow a few times
10. Hover over the selected tag value

## Expected Result:
There should be a hover effect visible for the selected category as is the case for the selected tag

## Actual Result:
There is no hover effect for the selected category after the focus has shifted from it which is different from the behavior seen for the selected independent tag value

## Workaround:
n/a

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

https://github.com/Expensify/App/assets/93399543/24fb1774-98cd-4585-a4bd-e3b00230842d

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by kbecciv** [comment]
We think that this bug might be related to #wave-collect - Release 1

**Comment by bernhardoj** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
There is no hover effect when hovering selected category.

### What is the root cause of that problem?
The category picker page uses SelectionList. The BaseListItem won't show the hover effect if the item is selected.
https://github.com/Expensify/App/blob/51c039d6632d49cacf4334d88d8276e9c8509fec/src/components/SelectionList/BaseListItem.tsx#L62

That's why there is no hover effect on the selected item even after we move away the item focus.

### What changes do you think we should make in order to solve the problem?

Instead of check whether the item is selected or not, we should check whether the item is focused or not.
```
hoverStyle={!item.isDisabled && !isFocused && styles.hoveredComponentBG}
```

_or just remove the selected condition_

**Comment by lschurr** [comment]
Hi @jayeshmangwani! Looks like we have a few proposals already. Could you please review when you get a chance? Thanks!

**Comment by jayeshmangwani** [comment]
Thanks for the proposals @bernhardoj @Krishna2323. I also feel we should remove the `selected` condition, but that was added in this [PR](https://github.com/Expensify/App/pull/37199/) and this [commit](https://github.com/Expensify/App/pull/37199/commits/7257a0f4bbe8d2ebbb059ce530a40956d6cf5ad7) intentionally,
For the output, let's confirm with the design team.

**Comment by jayeshmangwani** [comment]
The existing behaviour for the list items is if the list item is selected, we will not show the default hover effect (`theme.hoveredComponentBG`). Instead, the list uses the passed background color; 

For this issue, the proposal is to remove the condition `isSelected` from the ListItem. This will change the behaviour for the categories list, Tags list and Taxes list:

Right now, on the workspace categories page, when we select any category, it will show the background color to `theme.activeComponentBG`, and then hover over other list items and again hover over to the selected category is not change the background color.

After removing the `isSelected` conditions, the categories list's behaviour will be like this:

If the List item is selected, then the background color will be `theme.activeComponentBG`; if hovering  over the selected item, then the background color will be `theme.hoverComponentBG`

Adding videos below, before and after of this change:

<details>
  <summary>It looks like this: </summary>
  

https://github.com/Expensify/App/assets/35371050/5a15b855-6c34-4aaa-b20b-9facb7c54899


 </details>
    
  <details>
  <summary>After this change, it will look like this:</summary>
  
  https://github.com/Expensify/App/assets/35371050/0594b874-53b9-4745-955b-a0c8c0d26aef


  </details>


@shawnborton We will need your suggestions on whether we should remove the condition of the list selected.

**Comment by jayeshmangwani** [comment]
@Krishna2323 I have checked your proposal, but I feel that we should remove the is selected condition. Let's wait for the design team, and if we don’t want to change the existing behaviour of the categories list, then we can go with other proposals.

**Comment by dannymcclain** [comment]
>I'm not super convinced that's a major problem, but if we were to solve it, I would just make sure even a selected row would get our rowHover background color when you hover over it

I agree with Shawn. Not totally convinced it's a problem, but agree with this solution if we deem we should change it.

Also, why does the keyboard highlight use the "selected" color state instead of just the "hover" color state?

**Comment by jayeshmangwani** [comment]
>Also, why does the keyboard highlight use the "selected" color state instead of just the "hover" color state?

@dannymcclain I'm not sure why it was added, but I am tagging the @luacmartins  and @burczu as this is coming from this [PR](https://github.com/Expensify/App/pull/37199/files) and  In that PR, we are setting it not to show the hover background color if the list item is selected.

**Comment by luacmartins** [comment]
We assumed that was the expected behavior. Happy to change it if we decide to show the hover effect on a selected item.

**Comment by jayeshmangwani** [comment]
@luacmartins Then we can go with the @bernhardoj's [proposal](https://github.com/Expensify/App/issues/39716#issuecomment-2039886372) of remove the selected condition 

🎀 👀 🎀 C+ reviewed

**Comment by luacmartins** [comment]
It seems like we're still discussing if this is a problem or not. Waiting on @Expensify/design's decision here.

**Comment by shawnborton** [comment]
I guess the problem I see with giving a selected row a hover style is that you are kind of already hovering over the row when you go to select it... so I have no idea how that would work - do you instantly get the selected BG color when it's selected, but as soon as the mouse moves we change the BG color?

So that leads me to think we should just leave this one alone, and not add any hover BG colors to select rows. Thoughts @Expensify/design ?

**Comment by dannymcclain** [comment]
>So that leads me to think we should just leave this one alone, and not add any hover BG colors to selected rows.

I am totally down to leave it alone.

**Comment by luacmartins** [comment]
Ok, so it sounds like we can just close this issue then?

**Comment by dubielzyk-expensify** [comment]
I actually think hover on selected item also might looking a bit odd. But yeah, I'm with the designers on this

**Comment by shawnborton** [comment]
> However, the issue arises when there is no hover style applied to an item that doesn't have any background and is selectable. 

Yup, that makes sense. I'd be down to fix that particular case.

**Comment by dannymcclain** [comment]
I'm fine with fixing that case too, but I don't feel too strongly.

**Comment by dubielzyk-expensify** [comment]
I actually think what we have here is working well. If there's already a background on a selected item, then don't do a hover style. If there isn't one, then use it. So I'm okay with this

**Comment by shawnborton** [comment]
Okay cool, sounds like design team agrees: let's move forward with @Krishna2323's idea [here](https://github.com/Expensify/App/issues/39716#issuecomment-2046811946). @Krishna2323 - you might need to update your proposal to recapture the problem statement and solution though.

**Comment by lschurr** [comment]
Commenting to remove overdue since this one is still moving.

**Comment by jayeshmangwani** [comment]
@Krishna2323's [proposal](https://github.com/Expensify/App/issues/39716#issuecomment-2040221601) will work if we have to add a hover effect to only those selected list items that have no background color. @luacmartins , if we agree on [this](https://github.com/Expensify/App/issues/39716#issuecomment-2046811946), then we can proceed with the proposal.

**Comment by trjExpensify** [comment]
Easily done! :)
